### PR TITLE
Fixes #12734: Add documentation for the Centreon plugin

### PR DESCRIPTION
**Pull request by VictorQrt**
https://www.rudder-project.org/redmine/issues/12734

### CODE REVIEW

File: centreon/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+Plugin for Centreon and Rudder
+==============================
+
+This project is part of Rudder - IT automation and compliance made easy.
+See: http://rudder-project.org for more information.
+
+This project is also part of iTop - CMDB and ITSM open source software.
```
Comment (peckpeck) [inline centreon/README.md]: what ?

File: centreon/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+- Prerequisites :
+
+  - Install a Centreon server https://download.centreon.com
+    The server is not required to be running on the same host as the Rudder server.
+    However, this plugin has to be installed and running on the Rudder server.
+
+  - The Rudder server must have Python installed.
+
+- Generate the Rudder package : make
```
Comment (peckpeck) [inline centreon/README.md]: we are not in the root directory, say where we should run make

Comment (peckpeck) [inline centreon/README.md]: We should probably add that the rpkg file is available somewhere

File: centreon/README.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+- Install the package on the Rudder server : /opt/rudder/bin/rudderpkg install-file <plugin>.rpkg
+
+- Edit the configuration file /opt/rudder/etc/centreon.conf
```
Comment (peckpeck) [inline centreon/README.md]: what interesting values in this file what do they mean ?

File: centreon/README.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
```
Comment (peckpeck) [inline centreon/README.md]: what happens next ? 
what sould i do to check if the plugin is working ?

**Comment by VictorQrt** [comment]
Commit modified

File: centreon/README.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+- Prerequisites :
+
+  - Install a Centreon server https://download.centreon.com
+    The server is not required to be running on the same host as the Rudder server.
+    However, this plugin has to be installed and running on the Rudder server.
+
+  - The Rudder server must have Python installed.
+
+- Generate the Rudder package : cd into the centreon directory, then make. The .rpkg package file
```
Comment (peckpeck) [inline centreon/README.md]: if you don't have the binary rpkg file ...

File: centreon/README.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  will be generated in the current directory.
+
+- Install the package on the Rudder server : /opt/rudder/bin/rudderpkg install-file <plugin>.rpkg
```
Comment (peckpeck) [inline centreon/README.md]: isn't there a markdown syntax for commands ?

File: centreon/README.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+- Edit the configuration file /opt/rudder/etc/centreon.conf. This config file has to contain your Centreon
+  server's webservice API URL, the credentials to access it, and the name of the Centreon poller used to 
+  monitor Rudder nodes (Central by default). You also need to provide the Rudder API URL, and a token to access
+  it, which you can issue from the Rudder server webapp or CLI.
```
Comment (peckpeck) [inline centreon/README.md]: " from the Rudder server webapp or CLI" this is too generic, either say how to get it or don't say anything.
The best would be a pointer to a Rudder documentation url that already has the information.

File: centreon/README.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+
+Usage
+-----
+
+Now that the plugin is installed, you can check if it is working by adding new Rudder nodes. They will be added
+to you list of Centreon hosts (and removed on deletion of the Rudder node).
+
+Centreon comes with the possiblity to add monitoring templates to its hosts. You can create templates in
+Centreon or check for available ones. Once you know the name of a template suitable for a Rudder node/nodegroup,
+you can add it using the "Monitoring template" generic method from the technique editor.
```
Comment (peckpeck) [inline centreon/README.md]: How long to i need to wait to make sure the template is applied ?

Comment (peckpeck) [reply centreon/README.md]: a word about the cron/daemon that process those templates and how it works ?

File: centreon/README.md [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+- Edit the configuration file /opt/rudder/etc/centreon.conf. This config file has to contain your Centreon
+  server's webservice API URL, the credentials to access it, and the name of the Centreon poller used to 
+  monitor Rudder nodes (Central by default). You also need to provide the Rudder API URL, and a token to access
+  it, which you can issue from the Rudder server webapp or CLI.
+
+Usage
+-----
+
+Now that the plugin is installed, you can check if it is working by adding new Rudder nodes. They will be added
+to you list of Centreon hosts (and removed on deletion of the Rudder node).
```
Comment (peckpeck) [inline centreon/README.md]: Add a sentence to briefly describe how it works : post-hook, api call from rudder server ...

**Comment by VictorQrt** [comment]
Commit modified

File: centreon/README.md [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+- Edit the configuration file /opt/rudder/etc/centreon.conf. This config file has to contain your Centreon
+  server's webservice API URL, the credentials to access it, and the name of the Centreon poller used to
+  monitor Rudder nodes (Central by default). You also need to provide the Rudder API URL, and a token to access
+  it, which you will need to issue : see https://www.rudder-project.org/rudder-api-doc/#api-_-Authentication
+
+Usage
+-----
+
+Now that the plugin is installed, you can check if it is working by adding new Rudder nodes. They will be added
+to you list of Centreon hosts (and removed on deletion of the Rudder node).
```
Comment (peckpeck) [inline centreon/README.md]: to you*r* list

File: centreon/README.md [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+
+- Edit the configuration file /opt/rudder/etc/centreon.conf. This config file has to contain your Centreon
+  server's webservice API URL, the credentials to access it, and the name of the Centreon poller used to
+  monitor Rudder nodes (Central by default). You also need to provide the Rudder API URL, and a token to access
+  it, which you will need to issue : see https://www.rudder-project.org/rudder-api-doc/#api-_-Authentication
+
+Usage
+-----
+
+Now that the plugin is installed, you can check if it is working by adding new Rudder nodes. They will be added
```
Comment (peckpeck) [inline centreon/README.md]: automatically added

File: centreon/README.md [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+to you list of Centreon hosts (and removed on deletion of the Rudder node).
+
+This is achieved by Rudder's post-node-acceptance/deletion hooks, which are calling the plugin with the appropriate
+options on such events.
+
+Centreon comes with the possiblity to add monitoring templates to its hosts. You can create templates in
+Centreon or check for available ones. Once you know the name of a template suitable for a Rudder node/nodegroup,
```
Comment (peckpeck) [inline centreon/README.md]: suitable template

File: centreon/README.md [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+you can add it using the "Monitoring template" generic method from the technique editor.
+
+In order to configure the monitoring of the Rudder nodes, you can provide Centreon with parameters associated to its
+hosts. The "Monitoring parameter" generic method achieves this.
+
+The data will be sent hourly to the Centreon server. This can be changed by editing the /etc/cron.d/centreon-rudder file.
+This operation can be manually executed by running
```
Comment (peckpeck) [inline centreon/README.md]: running where ?

**Comment by VictorQrt** [comment]
Commit modified

**Pull request merged**